eva/io_descriptors/eva_data_types.py

```diff
@@ -0,0 +1,144 @@
+    ) -> None:
+        super().__init__(
+            name=name,
+            type=ColumnType.NDARRAY,
+            is_nullable=is_nullable,
+            array_type=type,
+            array_dimensions=dimensions,
+        )
+
+    def check_type(self, input_object) -> bool:
+        if self.dtype:
+            if self.dtype == "int32":
+                return isinstance(input_object, np.ndarray) and (
+                    input_object.dtype == np.int32
+                )
+            elif self.dtype == "float16":
+                return isinstance(input_object, np.ndarray) and (
+                    input_object.dtype == np.float16
+                )
+            elif self.dtype == "float32":
+                return isinstance(input_object, np.ndarray) and (
+                    input_object.dtype == np.float32
+                )
+
+        elif not self.dtype:
+            return isinstance(input_object, np.ndarray)
+
+    def check_shape(self, input_object) -> bool:
+        if self.shape:
+            if input_object.shape != self.shape:
+                return False
+
+        return True
+
+    def arg_name(self):
+        return "NumpyArray"
+
+
+class PyTorchTensor(IOColumnArgument):
+    """EVA data type for PyTorch Tensor"""
+
+    def __init__(
+        self,
+        name: str,
+        is_nullable: bool = False,
+        type: NdArrayType = None,
+        dimensions: Tuple[int] = None,
+    ) -> None:
+        super().__init__(
+            name=name,
+            type=ColumnType.NDARRAY,
+            is_nullable=is_nullable,
+            array_type=type,
+            array_dimensions=dimensions,
+        )
+
+    def check_type(self, input_object) -> bool:
+        if self.array_type:
+            if self.array_type == "int32":
+                return isinstance(input_object, torch.Tensor) and (
+                    input_object.dtype == torch.int32
+                )
+            elif self.array_type == "float16":
+                return isinstance(input_object, torch.Tensor) and (
+                    input_object.dtype == torch.float16
+                )
+            elif self.array_type == "float32":
+                return isinstance(input_object, torch.Tensor) and (
+                    input_object.dtype == torch.float32
+                )
+
+        elif not self.array_type:
+            return isinstance(input_object, torch.Tensor)
+
+    def check_shape(self, input_object) -> bool:
+        if self.shape:
+            if input_object.shape != self.shape:
+                return False
+
+        return True
+
+    def arg_name(self):
+        return "PyTorch Tensor"
+
+
+class PandasDataframe(IOArgument):
+    """EVA data type for Pandas Dataframe"""
+
+    def __init__(self, columns, shape, enforce_columns, enforce_shape) -> None:
+        super().__init__()
+        self.columns = columns
+
+        self.type = ColumnType.NDARRAY
+        self.array_type = NdArrayType.ANYTYPE
+        self.is_nullable = False
```

Comment:
`type`, `array_type`, and `is_nullable` do not apply to a dataframe IMO. We probably need a function that converts the entire dataframe to a list of NumpyArguments assigning the correct class.